lib/checks/aria/aria-allowed-role.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * Implements allowed roles defined at:
+ * https://www.w3.org/TR/html-aria/#docconformance
+ * https://www.w3.org/TR/SVG2/struct.html#implicit-aria-semantics
+ */
+const { allowImplicit = true, ignoredTags = [] } = options || {};
+
+const tagName = node.tagName.toLowerCase();
+
+// check if the element should be ignored, by an user setting
+if (ignoredTags && ignoredTags.indexOf(tagName) > -1) {
+	return true;
+}
+
+const roleSegments = axe.commons.aria.getRoleSegments(node);
+const implicitRole = axe.commons.aria.implicitRole(node);
+
+/**
+ * Helper object with functions to computed is role allowed for a given tag
```

Comment:
I understand what is meant here, but the language is quite unclear. Maybe the result of surgical edits? 😄